### PR TITLE
Fixing warehouse services

### PR DESCRIPTION
**Pull request by beatrizleon**
### Description

When launching the warehouse services, I got an error connecting to the warehouse which was due to not having the appropriate try/catch and also missing the conn->connect() call. With this pr, services are advertised correctly.

### Checklist
- [ ] **Required**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] I tried on Kinetic but I think it should be cherry-picked to other current ROS branches (Indigo, Jade)

Thank you!

### CODE REVIEW

File: moveit_ros/warehouse/warehouse/src/warehouse_services.cpp [hunk 1]
```diff
@@ -133,11 +133,26 @@ int main(int argc, char** argv)
...
-  conn->setParams(host, port, 5.0);
+  try
+  {
+    conn = moveit_warehouse::loadDatabase();
+    conn->setParams(host, port, 5.0);
+
+    ROS_INFO("Connecting to warehouse on %s:%d", host.c_str(), port);
+    if (!conn->connect())
+    {
+      ROS_ERROR("Failed to connect to DB");
```
Comment (jrgnicho) [inline moveit_ros/warehouse/warehouse/src/warehouse_services.cpp]: It'd be nice to print the host and port on this error message. I'm surprised it ever worked without the call to **connect()**

**Comment by beatrizleon** [comment]
@jrgnicho thanks for reviewing it. I added info to the comment.

**Comment by v4hn** [comment]
Thanks for the contribution @beatrizleon !
The current state of the warehouse is not too nice I suppose..

**Comment by beatrizleon** [comment]
@v4hn thanks for the merge! Do you have an estimate on when these changes will be released to the deb package?

**Comment by v4hn** [comment]
On Wed, Mar 22, 2017 at 05:31:47AM -0700, Beatriz Leon wrote:
> @v4hn thanks for the merge! Do you have an estimate on when these changes will be released to the deb package?

@130s is in charge of the release process.

We had a release just two weeks ago (https://github.com/ros-planning/moveit/commit/f3b6a045cf415af61ba48e750f0caceb6e26b45a).
Personally, I would like to wait with the next one until we accumulated some more fixes or other interesting additions (maximum another month).
If you need this fix sooner or another maintainer wants a release sooner, we could have another patch release though, I guess.

-- 
Michael Görner, M.Sc. Cognitive Science, PhD Student
Universität Hamburg
Faculty of Mathematics, Informatics and Natural Sciences
Department of Informatics
Group Technical Aspects of Multimodal Systems

Vogt-Kölln-Straße 30
D-22527 Hamburg

Room: F-315
Phone: +49 40 42883-2432

**Pull request merged**